### PR TITLE
Document the drive classes and their commonly used features

### LINKED ISSUE #1109: Document the drive classes
It should cover the following:

1. What drive inverse kinematics classes are available (DifferentialDrive, MecanumDrive)
2. What inverse kinematic methods each class provides
3. What RobotDriveBase.setDeadband() does (adds a deadband to joystick values while rescaling the passthrough region so there isn't an output discontinuity; also note that users can square the input or apply other transformations to the input and the deadband will compose with it)
4. What RobotDriveBase.setMaxOutput() does

### PR DESCRIPTION
**Pull request by jasondaming**
Closes #1109

### CODE REVIEW

**Comment by Daltz333** [comment]
I don't believe this fully solves the issue. @calcmogul

**Comment by Daltz333** [comment]
I believe Tyler wanted examples of
```
.arcadeDrive()
.tankDrive()
.curvatureDrive()
```

```
driveCartesian()
drivePolar()
```

**Comment by jasondaming** [comment]
You mean like the example code in the article:
```
public void teleopPeriodic() {
    // Tank drive with a given left and right rates
    myDrive.tankDrive(leftStick.getY(), rightStick.getY());

    // Arcade drive with a given forward and turn rate
    myDrive.arcadeDrive(driveStick.getY(),driveStick.getX());

    // Curvature drive with a given forward and turn rate, as well as a quick-turn button
    myDrive.curvatureDrive(driveStick.getY(), driveStick.getX(), driveStick.getButton(1));
}```

**Comment by jasondaming** [comment]
And then this one:
```public void teleopPeriodic() {
    m_robotDrive.driveCartesian(m_stick.getX(), m_stick.getY(), m_stick.getZ());
    m_robotDrive.drivePolar(m_stick.getX(), m_stick.getY(), m_stick.getZ());
}```

File: source/docs/software/actuators/wpi-drive-classes.rst [hunk 1]
```diff
@@ -27,19 +27,25 @@ When viewed from the top, the rollers on a mecanum drivetrain should form an 'X'
...
 
-By default, the Differential Drive class applies an input deadband of .02. This means that input values with a magnitude below .02 (after any squaring as described above) will be set to 0. In most cases these small inputs result from imperfect joystick centering and are not sufficient to cause drivetrain movement, the deadband helps reduce unnecessary motor heating that may result from applying these small values to the drivetrain. To change the deadband, use the setDeadband() method.
+Input Deadband
+^^^^^^^^^^^^^^
+
+By default, the Differential Drive class applies an input deadband of .02. This means that input values with a magnitude below .02 (after any squaring as described above) will be set to 0. In most cases these small inputs result from imperfect joystick centering and are not sufficient to cause drivetrain movement, the deadband helps reduce unnecessary motor heating that may result from applying these small values to the drivetrain. To change the deadband, use the `setDeadband()` method.
```
Comment (calcmogul) [inline source/docs/software/actuators/wpi-drive-classes.rst:43]:
```suggestion
By default, the Differential Drive class applies an input deadband of 0.02. This means that input values with a magnitude below 0.02 (after any squaring as described above) will be set to 0. In most cases these small inputs result from imperfect joystick centering and are not sufficient to cause drivetrain movement, the deadband helps reduce unnecessary motor heating that may result from applying these small values to the drivetrain. To change the deadband, use the `setDeadband()` method.
```

**Review by calcmogul — CHANGES_REQUESTED** [review]

**Comment by jasondaming** [comment]
@calcmogul / @Daltz333 is there anything else that needs to be done on this PR?

**Comment by Daltz333** [comment]
I was waiting on Tyler's review to make sure it adequately addressed his comments

**Review by calcmogul — APPROVED** [review]

**Review by Daltz333 — APPROVED** [review]

**Pull request merged**